packages/wallet-sdk/src/sign/scw/SCWSigner.ts

```diff
@@ -269,4 +297,62 @@ export class SCWSigner implements Signer {
     }
     return true;
   }
+
+  private async addAddress(request: RequestArguments) {
+    const state = SubAccount.getState();
+    if (state.account) {
+      return state.account;
+    }
+
+    await this.communicator.waitForPopupLoaded?.();
+    let signer = get(request, 'params[0].signer') as string;
+    if (!state.getSigner) {
+      throw standardErrors.rpc.invalidParams('no signer provided');
+    }
+    const account = await state.getSigner();
+    if (!signer && account) {
+      signer = account.publicKey;
+    }
+    if (!Array.isArray(request.params)) {
+      throw standardErrors.rpc.invalidParams();
+    }
+    request.params[0].signer = signer;
+
+    const response = await this.sendRequestToPopup(request);
+
+    // Only store the sub account information after the popup has been closed and the
+    // user has confirmed the creation
+    SubAccount.setState({
+      account: response as SubAccountInfo,
+    });
+    return response;
+  }
+
+  private shouldRequestUseSubAccountSigner(request: RequestArguments) {
+    const sender = getSenderFromRequest(request);
+    // if the sender is undefined, it means the application did not provide a sender
+    // in this case, we assume the request is for the active sub account (if present)
+    // if the sender is defined, we check if it is the same as the active sub account
+    // if not, we use the root account signer
+    const state = SubAccount.getState();
+    return (
+      (state.account && sender === undefined) || (state.account && state.account.address === sender)
+    );
+  }
+
+  private async sendRequestToSubAccountSigner(request: RequestArguments) {
+    const state = SubAccount.getState();
+    if (!state.account) {
+      throw standardErrors.provider.unauthorized('No active sub account');
```

Comment:
I noticed a few checks for presence. It could be nice to abstract this.

```typescript
function assertPresence<T>(value: T | null | undefined, error: (message: string) => Error, message: string): asserts value is T {
  if (value === null || value === undefined) {
    throw error('Value must be present');
  }
}

// Example usage
// Assert the presence of the value
assertPresence(state.account, standardErrors.provider.unauthorized);
```